clang/lib/Driver/Driver.cpp

```diff
@@ -922,6 +947,8 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
                    IsAMDOffloadArch(StringToOffloadArch(
                        getProcessorFromTargetID(*AMDTriple, Arch)))) {
           DerivedArchs[AMDTriple->getTriple()].insert(Arch);
+        } else if (SPIRVTriple && Arch == (*SPIRVTriple).str()) {
```

Comment:
What is `Arch` here exactly? My understanding is that SPIR-V has no concept of `-mcpu` arguments.

Reply:
Yeah so here this is a case where SPIR-V is an outlier so stuff gets weird. This code gets hit when `-fopenmp` is passed, `-fopenmp-targets` is not passed, and `--offload-arch` is passed and we need to deduce the toolchain based on the offload arch. I implemented it such that you can pass `spirv64-intel` to `--offload-arch` and it will use the SPIR-V OpenMP toolchain. Arch will be `spirv64-intel` so we go inside the `if` and everything works.

Another option is that we could not allow the `spirv64-intel` option to `--offload-arch` to deduce to the SPIR-V OpenMP toolchain in this case, so basically you can't use `--offload-arch` with this toolchain. Maybe that's philosophically better. 

Let me know what you think.

Reply:
This is related to discussions I've had with @AlexVlx who is working on SPIR-V for HIP, but I strongly believe that we should deprecate `-fopenmp-targets=` and `--offload-arch=` should not be used for SPIR-V. We should fix-up the `--offload=` option to take the list of toolchains directly, and `--offload-arch=` should be used as a shorthand infereence. Then have `-Xoffload-<triple> -arg` to control specific arguments to toolchains.

Reply:
Sure, if that's is the direction we're going I can just remove this code so `--offload-arch` errors and you need `-fopenmp-targets=spirv64-intel` for now until the `--offload` option is updated.